donkeycar/parts/lidar.py

```diff
@@ -13,6 +16,257 @@
+                        # time: time in seconds as a float
+                        # scan:  full scan this measurement belongs to
+                        #        as an integer
+                        # index: index within full scan as an integer
+                        #
+                        # Note: The scan:index pair represents a natural key
+                        #       identifying the measurement. This driver
+                        #       maintains a circular buffer of measurements
+                        #       that represent the most recent 360 degrees
+                        #       of measurements.  This list may include
+                        #       measurements from the current full scan and
+                        #       from the previous full scan.  So if
+                        #       run_threaded() is called rapidly
+                        #       (faster than the scan rate of the lidar),
+                        #       then the returned scans will have some new
+                        #       values and some values that may have been
+                        #       seen in the previous scan.  The scan:index
+                        #       pair can be used to
+                        #       to 'diff' scans to see which measurements
+                        #       are are 'new' and which measurements are
+                        #       shared between scans.
+                        #
+                        #       The time at which the measurement was
+                        #       aquired is also included. In a moving
+                        #       vehicle older measurements are less
+                        #       relevant; the time field can be used to
+                        #       filter out older measurements or to
+                        #       visualize them differently (fading them out
+                        #       perhaps). It may also help when using a
+                        #       kinematic model to adjust for movement
+                        #       of the lidar when attached to a vehicle.
+                        #
+                        measurement = (distance, angle, now,
+                                        self.full_scan_count, self.full_scan_index)
+                        
+                        # grow buffer if necessary, otherwise overwrite
+                        if self.measurement_index >= len(self.measurements):
+                            self.measurements.append(measurement)
+                            self.measurement_count = self.measurement_index + 1
+                        else:
+                            self.measurements[self.measurement_index] = measurement  # noqa
+                        self.measurement_index += 1
+                        self.full_scan_index += 1
+                            
+            except serial.serialutil.SerialException:
+                logger.info('SerialException from RPLidar.')
```

Comment:
This probably should be a `logger.error(...)`

Reply:
And so it should be, and so it is.